sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4306,8 +4306,14 @@ pi_result piProgramBuild(pi_program Program, pi_uint32 NumDevices,
   // check now for unresolved symbols.
   ze_result_t ZeResult = checkUnresolvedSymbols(ZeModule, &Program->ZeBuildLog);
   if (ZeResult == ZE_RESULT_ERROR_MODULE_LINK_FAILURE) {
+    // remove ZeModule that is associated with the failed program
+    ZE_CALL_NOCHECK(zeModuleDestroy, (ZeModule));
+    ZeModule = nullptr;
     return PI_BUILD_PROGRAM_FAILURE;
   } else if (ZeResult != ZE_RESULT_SUCCESS) {
+    // remove ZeModule that is associated with the failed program
+    ZE_CALL_NOCHECK(zeModuleDestroy, (ZeModule));
```

Comment:
Please restructure if/else such that there is only one call to zeModuleDestroy

Reply:
Done. Thanks.